### PR TITLE
Bug 2015549: Fix localization misses in TablePagination and Metrics components

### PR DESCRIPTION
**Pull request by jcaianirh**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2015549

Note: The column headings are a result from a query and are dynamic data that aren't localized. That being said, there were a few other localization issues on the page that I fixed. The strings to the PF Pagination component needed to be localized and passed in. 

In the PF PaginationOptionsMenu component I issued a bug to allow the ofWord prop to be used in the '1-4 of 4' string in the screenshot below. We already pass that prop in, so when pf fixes that issue we should get the localized string for free: https://github.com/patternfly/patternfly-react/issues/6483

Before:
Screen Shot 2021-10-19 at 5 12 46 PM

After:
Screen Shot 2021-10-19 at 5 41 27 PM

### CODE REVIEW

**Comment by kyoto** [comment]
Thank you @jcaianirh! It seems I missed a few translations in this file 😬 

For the `ofWord` case, I think it will cause a couple of problems.
1. The desired translation will not necessary fit the `X $ofWord Y` pattern for all languages. For example, in Japanese you might translate "5-10 of 20" as "20件中 5~10件".
2. The translation team will need to translate the word `of` on its own, but `of` has different translations depending on the context. They can look at the context in this case to pick the right translation, but in the future another page may want a translation for `of` and pick up a translation that likely won't fit the new context.

So, I think we should be translating the whole `X of Y` string with X and Y as parameters. I think this is probably the best approach.

Alternatively, maybe we could get away with using `/` (not translated) instead of `of` (e.g. `5-10 / 20` for all languages).

**Comment by jcaianirh** [comment]
@kyoto @rebeccaalpert @megan-hall updated the solution for the "1 - 9 of 10" localization issue with a template that can be passed into the pf pagination component that provides context to the translators for the entire text string being rendered. This takes the place of using "1 - 9 / 10" or passing in 'of' to the component without context. Let me know if you have any questions.

**Review by rebeccaalpert — APPROVED** [review]
Looks good to me! Thanks for working towards a more context-full solution.

File: frontend/public/components/monitoring/table-pagination.tsx [hunk 1]
```diff
@@ -30,6 +44,21 @@ const TablePagination = ({
...
+        itemsPerPage: t('public~Items per page'),
+        perPageSuffix: t('public~per page'),
+        toFirstPage: t('public~Go to first page'),
+        toPreviousPage: t('public~Go to previous page'),
+        toLastPage: t('public~Go to last page'),
+        toNextPage: t('public~Go to next page'),
+        optionsToggle: t('public~Items per page'),
+        currPage: t('public~Current page'),
+        paginationTitle: t('public~Pagination'),
+        ofWord: t('public~of'),
```
Comment (kyoto) [inline frontend/public/components/monitoring/table-pagination.tsx:60]: Can `ofWord` now be removed from here?

Comment (kyoto) [reply frontend/public/components/monitoring/table-pagination.tsx:60]: Thanks for the explanation. It's unfortunate, but I'm not sure what a better solution would be, so probably keeping `ofWord` is reasonable.

Comment (kyoto) [reply frontend/public/components/monitoring/table-pagination.tsx:60]: I discussed with @rebeccaalpert and I don't think there is a better solution with our current i18n approach, so I think this is OK to merge.

**Pull request merged**